src/main/java/org/sonarlint/cli/Options.java

```diff
@@ -58,24 +60,38 @@ public static Options parse(String[] args) throws ParseException {
       } else if ("-X".equals(arg) || "--debug".equals(arg)) {
         options.verbose = true;
 
-      } else if ("--report-dir".equals(arg)) {
+      } else if ("--html-report".equals(arg)) {
         i++;
         if (i >= args.length) {
-          throw new ParseException("Missing argument for option --report-dir", i);
+          throw new ParseException("Missing argument for option " + arg, i);
         }
-        options.reportDir = args[i];
+        options.htmlReport = args[i];
 
-      } else if ("--report-name".equals(arg)) {
+      } else if ("--charset".equals(arg)) {
         i++;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor the code in order to not assign to this loop counter from within the loop body. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AForLoopCounterChangedCheck)